Add xdr extension

ulrichSchreiner opened the pull request:
add command for cortex-agent plugin

ulrichSchreiner committed:
fix dependency overwrite and repair code

ulrichSchreiner committed:
set repleacment

ulrichSchreiner committed:
set statuscode field only if it is set, otherwise leave it nil

ulrichSchreiner committed:
Merge branch 'fix-dependency' into add-xdr-extension

ulrichSchreiner committed:
some commands for xdr config

ulrichSchreiner committed:
add noproxy flag

ulrichSchreiner committed:
Merge branch 'master' into add-xdr-extension

ulrichSchreiner committed:
Merge branch 'master' into add-xdr-extension

ulrichSchreiner committed:
use latest cloud-go

ulrichSchreiner committed:
merge main

ulrichSchreiner committed:
tidy go.mod

ulrichSchreiner committed:
Merge branch 'master' into add-xdr-extension

ulrichSchreiner committed:
add cluster tenant to config

ulrichSchreiner committed:
do not set tenant in client

ulrichSchreiner committed:
merge main

ulrichSchreiner committed:
Merge branch 'master' into add-xdr-extension

Gerrit91 commented on cmd/cluster_xdr.go:
```diff
@@ -0,0 +1,101 @@
+	genericcli.Must(clusterXdrCmd.RegisterFlagCompletionFunc("cluster-id", c.comp.ClusterListCompletion))
+
+	configureCmd.Flags().Bool("disabled", false, "disables the entire xdr functionality")
+	configureCmd.Flags().String("distributionid", "", "the distribution id for the xdr configuration")
```
I think this can only be configured by provider tenants anyway? So maybe this should be contained in the command description before people start to play around with it and don't know what to do with it?

ulrichSchreiner committed:
change command description

the configuration for the XDR extension can only be set/changed by a provider tenant user
